packages/trader/src/Modules/Contract/Components/LastDigitPrediction/last-digit-prediction.tsx

```diff
@@ -51,21 +68,23 @@ const LastDigitPrediction = ({
         9: { left: 6 + dimension * 4, top: 8 },
     });
 
-    const handleSelect = digit_value => {
+    const handleSelect = (digit_value: number) => {
         if (!isSelectableDigitType()) return;
         if (digit_value !== selected_digit && typeof onDigitChange === 'function') {
             onDigitChange({ target: { name: 'last_digit', value: digit_value } });
         }
     };
 
-    const getBarrier = num => {
-        const barrier_map = {
-            DIGITMATCH: val => val === barrier,
-            DIGITDIFF: val => val !== barrier && !isNaN(barrier),
-            DIGITOVER: val => val > barrier,
-            DIGITUNDER: val => val < barrier,
-            DIGITODD: val => val % 2,
-            DIGITEVEN: val => !(val % 2),
+    const getBarrier = (num: number | null): number | null => {
+        const barrier_map: {
+            [key: string]: (val: number | null) => boolean;
+        } = {
+            DIGITMATCH: (val: number | null) => val === barrier,
+            DIGITDIFF: (val: number | null) => val !== barrier && barrier !== null && !isNaN(barrier),
+            DIGITOVER: (val: number | null) => val !== null && barrier !== null && val > barrier,
+            DIGITUNDER: (val: number | null) => val !== null && barrier !== null && val < barrier,
```

Comment:
```suggestion
            DIGITUNDER: (val: number | null) => val && barrier && val < barrier,
```
let's try shortening it like this for all of these lines🙂